internal/controller/mandatorymodule/deletion_controller.go

```diff
@@ -136,12 +136,16 @@ func (r *DeletionReconciler) removeManifests(ctx context.Context, manifests []v1
 	return nil
 }
 
-func filterManifestsByAnnotation(manifests []v1beta2.Manifest,
-	annotationKey, annotationValue string,
+func filterManifestsByFQDNAndVersion(manifests []v1beta2.Manifest,
+	fqdn, moduleVersion string,
 ) []v1beta2.Manifest {
 	filteredManifests := make([]v1beta2.Manifest, 0)
```

Comment:
Thanks for the update Nesma, Just one last point. I believe we have large manifests items to delete and we know the exact numbers. Maybe it make sense to allocate the capacity of the slice, for performance reason. But if there s few.
I suggest: 
`filteredManifests := make([]v1beta2.Manifest, 0)` to `filteredManifests := make([]v1beta2.Manifest, 0, lent(manifests))`